src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -818,11 +918,61 @@ TEST_F(ControllerScriptEngineLegacyTest, convertCharsetAllCharset) {
     }
 }
 
+TEST_F(ControllerScriptEngineLegacyTest, JavascriptPlayerProxy) {
+    QMap<QString, QString> expectedValues = {
+            std::pair("artist", "Test Artist"),
+            std::pair("title", "Test title"),
+            std::pair("album", "Test Album"),
+            std::pair("albumArtist", "Test Album Artist"),
+            std::pair("genre", "Test genre"),
+            std::pair("composer", "Test Composer"),
+            std::pair("grouping", ""),
+            std::pair("year", "2011"),
+            std::pair("trackNumber", "07"),
+            std::pair("trackTotal", "60")};
+
+    const auto keys = expectedValues.keys();
+    const QString jsValues = std::accumulate(keys.cbegin(),
+            keys.cend(),
+            QString(),
+            [](const QString first, const QString last) -> QString {
+                return first + "'" + last + "'" + ",";
+            });
+
+    const auto code = QString(
+            "var result = {};"
+            "var player = engine.getPlayer('[Channel1]');"
+            "for(const name of [%1]) {"
+            "    player[`${name}Changed`].connect(newValue => {"
+            "        result[name] = newValue;"
+            "    });"
+            "};")
+                              .arg(jsValues);
+    EXPECT_TRUE(evaluateAndAssert(code));
+
+    TrackPointer pTrack1 = getOrAddTrackByLocation(getTestDir().filePath("id3-test-data/all.mp3"));
+    auto deck = m_pPlayerManager->getDeck(1);
+    deck->slotLoadTrack(pTrack1,
+#ifdef __STEM__
+            mixxx::StemChannelSelection(),
+#endif
+            false);
+    m_pEngine->process(1024);
+    while (!deck->getEngineDeck()->getEngineBuffer()->isTrackLoaded()) {
+        QTest::qSleep(100);
+    }
+    processEvents();
+    for (auto [key, value] : expectedValues.asKeyValueRange()) {
+        EXPECT_QSTRING_EQ(value, qjsvalue_cast<QString>(evaluate(QString("player.%1").arg(key))));
+        EXPECT_QSTRING_EQ(value, qjsvalue_cast<QString>(evaluate(QString("result.%1").arg(key))));
+    }
```

Comment:
Is there a GoogleTest equivalent to [Python's `unittest.TestCase.subTest`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.subTest)? Couldn't find one.

Reply:
Not really. The decoding test has some code that tries to accomplish something similar, but I don't want you to reuse that. Looking at your test code however, I think you'll want to write a [Value-Parameterized Test](https://google.github.io/googletest/advanced.html#value-parameterized-tests).

Reply:
thats fine too, though a parametrized test wouldn't really be that much complicated IMO.